src/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php

```diff
@@ -103,13 +103,20 @@ public function load(array $configs, ContainerBuilder $container): void
         $formats = $this->getFormats($config['formats']);
         $patchFormats = $this->getFormats($config['patch_formats']);
         $errorFormats = $this->getFormats($config['error_formats']);
+        $docsFormats = $this->getFormats($config['docs_formats']);
 
         // Backward Compatibility layer
         if (isset($formats['jsonapi']) && !isset($patchFormats['jsonapi'])) {
             $patchFormats['jsonapi'] = ['application/vnd.api+json'];
         }
 
-        $this->registerCommonConfiguration($container, $config, $loader, $formats, $patchFormats, $errorFormats);
+        if (!isset($docsFormats['jsonopenapi'], $docsFormats['json'], $docsFormats['yamlopenapi'])) {
```

Comment:
What about this instead? See my other comment below.

```suggestion
        if ((!isset($formats) || isset($formats['json'])) && !isset($docsFormats)) {
```

IMHO this a a better BC layer.